### DISCUSSION TITLE
How to load pretrained weight for backbone in detectron2.modeling.backbone.resnet

### DISCUSSION THREAD

**Discussion by Frank-Y-Zheng** [facebookresearch/detectron2#3919] [Q&A]
I can only write 
`from detectron2.modeling.backbone.resnet import BasicStem, ResNet`
`import torch`
`model = ResNet(BasicStem(), ResNet.make_default_stages(101), out_features=None)`
but I can not find a way to load pretrained weight for the model. Can anyone help me?

**Comment by zhanghang1989** [comment]
The easiest way is using torchvision's resnet and weights.
detectron2 resnet weights only work when used as backbone.

https://github.com/facebookresearch/detectron2/blob/main/configs/COCO-Detection/fast_rcnn_R_50_FPN_1x.yaml#L3

**Reply by Frank-Y-Zheng** [reply]
Thank you. I do want to use it as backbone in detection network that write by myself. In fact, I can not find a efficient way to load pretrained backbone in code directly, can you help me with that?

**Reply by zhanghang1989** [reply]
You may write a custom backbone and load the weights inside the function.

https://github.com/facebookresearch/detectron2/blob/main/detectron2/modeling/backbone/resnet.py#L613
